src/main/java/edu/usf/cutr/grha/gtfs/GtfsWriterMain.java

```diff
@@ -0,0 +1,85 @@
+package edu.usf.cutr.grha.gtfs;
+
+import edu.usf.cutr.grha.model.ChicagoTncData;
+import org.onebusaway.gtfs.model.Agency;
+import org.onebusaway.gtfs.model.AgencyAndId;
+import org.onebusaway.gtfs.model.Route;
+import org.onebusaway.gtfs.model.Stop;
+import org.onebusaway.gtfs.serialization.GtfsWriter;
+
+import java.io.File;
+import java.util.List;
+
+public class GtfsWriterMain {
+
+    public String filePath;
+
+    public GtfsWriterMain(String filePath) {
+        this.filePath = filePath;
+    }
+
+    public void startWritingData(List<ChicagoTncData> chicagoTncDataList) {
+        GtfsWriter  writer = new GtfsWriter();
+        writer.setOutputLocation(new File(filePath));
+
+        //Creates a single fake agency in agency.txt for TNC trips
+        Agency agency = setFakeAgency();
+        writer.handleEntity(agency);
+
+        //Creates a single fake route in routes.txt that all TNC trips can be assigned to
+        writer.handleEntity(setFakeRoute(agency));
+
+        int counter = 0;
+        for (ChicagoTncData tncData: chicagoTncDataList) {
+
+            //Creates a bus stop (for stops.txt) for the origin and destination location
+            Stop originStop = setStop(tncData, agency, counter, true);
+            counter++;
+
+            writer.handleEntity(originStop);
+
+            Stop destinationStop = setStop(tncData, agency, counter, false);
+            counter++;
+
+            writer.handleEntity(destinationStop);
+
+        }
+
+        try {
+            writer.close();
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    public static Agency setFakeAgency() {
+        Agency agency = new Agency();
+        agency.setId(GtfsConstants.FAKE_AGENCY_ID);
+        agency.setName(GtfsConstants.FAKE_AGENCY_NAME);
+        agency.setUrl(GtfsConstants.FAKE_AGENCY_URL);
+        agency.setTimezone(GtfsConstants.AGENCY_TIMEZONE);
+        return agency;
+    }
+
+    public static Route setFakeRoute(Agency agency) {
+        Route route = new Route();
+        route.setId(AgencyAndId.convertFromString(agency.getId() + "_" + GtfsConstants.FAKE_ROUTE_ID));
```

Comment:
You don't need to prefix the IDs with the `agency.getId()` when writing a GTFS dataset as entity IDs are guaranteed to be unique within the GTFS dataset (you may have seen this practice elsewhere in OBA, but this is only needed internally in OBA when multiple GTFS are imported and entity IDs from multiple GTFS datasets might collide). So you can remove `agency.getId() + "_" + `. Same below.

Reply:
I tried setting the ID without the prefix and the agencyID but the dependency returned an exception saying Invalid agency ID.

Reply:
Hmmm, maybe that's a requirement in the onebusaway-gtfs-modules - I haven't used them to write GTFS before. That's fine if route.setId() requires it. Can you paste the exact error it gives without the prefix below?

Reply:
```
Exception in thread "main" java.lang.IllegalArgumentException: invalid agency-and-id: 1
        at org.onebusaway.gtfs.model.AgencyAndId.convertFromString(AgencyAndId.java:127)
        at edu.usf.cutr.grha.gtfs.GtfsWriterMain.setFakeRoute(GtfsWriterMain.java:66)
        at edu.usf.cutr.grha.gtfs.GtfsWriterMain.startWritingData(GtfsWriterMain.java:30)
        at edu.usf.cutr.grha.ProcessorMain.main(ProcessorMain.java:47)
```

Here 1 is the Fake route ID.

Reply:
Do you have to use AgencyAndId.convertFromString() here? Can you just set the route ID as a string directly?

Reply:
@barbeau I dont think so - The `setId` function requires an argument of type `agency-and-id` which is returned by the `AgencyAndId.convertFromString()`